File: api-net/Models/Tenant.cs
namespace LiftOtonom.Api.Models;

public class Tenant
{
    public long Id { get; set; }
    public string Name { get; set; } = "";
    public string Slug { get; set; } = "";
    public string? Phone { get; set; }
    public string? Email { get; set; }
    public string? Address { get; set; }
    public string? TaxNumber { get; set; }
    public string? TaxOffice { get; set; }
    public string? LogoUrl { get; set; }
    public string Plan { get; set; } = "trial";
    public DateTime? PlanExpiresAt { get; set; }
    public int SmsBalance { get; set; } = 100;
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public DateTime? DeletedAt { get; set; }

    public bool IsUsable() => IsActive && (PlanExpiresAt == null || PlanExpiresAt > DateTime.UtcNow);
}

File: api/app/Models/Tenant.php
<?php
// app/Models/Tenant.php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\SoftDeletes;

class Tenant extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'name', 'slug', 'phone', 'email', 'address', 'tax_number', 'tax_office',
        'logo_url', 'plan', 'plan_expires_at', 'sms_balance', 'is_active', 'settings',
    ];

    protected function casts(): array
    {
        return [
            'plan_expires_at' => 'datetime',
            'sms_balance'     => 'integer',
            'is_active'       => 'boolean',
            'settings'        => 'array',
        ];
    }

    public function users(): HasMany
    {
        return $this->hasMany(User::class);
    }

    public function subscription(): HasMany
    {
        return $this->hasMany(\App\Models\Subscription::class);
    }

    public function isActive(): bool
    {
        return $this->is_active && (
            $this->plan_expires_at === null || $this->plan_expires_at->isFuture()
        );
    }
}
